specification/confidentialledger/data-plane/Microsoft.ConfidentialLedger/stable/2022-05-13/confidentialledger.json

```
@@ -0,0 +1,949 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      }
+    },
+    "/app/users/{userId}": {
+      "delete": {
+        "operationId": "ConfidentialLedger_DeleteUser",
+        "x-ms-examples": {
+          "DeleteUser": {
+            "$ref": "./examples/ledger/DeleteUser.json"
+          }
+        },
+        "summary": "Deletes a user from the Confidential Ledger.",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/UserIdParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "The specified user is deleted."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      },
+      "get": {
+        "operationId": "ConfidentialLedger_GetUser",
+        "x-ms-examples": {
+          "GetUser": {
+            "$ref": "./examples/ledger/GetUser.json"
+          }
+        },
+        "summary": "Gets a user.",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/UserIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Details about the specified user.",
+            "schema": {
+              "$ref": "#/definitions/LedgerUser"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "ConfidentialLedger_CreateOrUpdateUser",
+        "x-ms-examples": {
+          "CreateOrUpdateUser": {
+            "$ref": "./examples/ledger/CreateOrUpdateUser.json"
+          }
+        },
+        "summary": "Adds a user or updates a user's fields.",
+        "description": "A JSON merge patch is applied for existing users",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/UserIdParameter"
+          },
+          {
+            "$ref": "#/parameters/UserDetailsParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "User created or updated successfully.",
+            "schema": {
+              "$ref": "#/definitions/LedgerUser"
+            }
+          },
```

If this operation can create or update a user, it should return a 201 when creating and 200 when updating:
```suggestion
          "200": {
            "description": "User updated successfully.",
            "schema": {
              "$ref": "#/definitions/LedgerUser"
            }
          },
          "201": {
            "description": "User created successfully.",
            "schema": {
              "$ref": "#/definitions/LedgerUser"
            }
          },
```

---

Unfortunately, we only return 200 right now. It's a PATCH call.